build.gradle

```diff
@@ -179,11 +179,14 @@ configure(opentelemetryProjects) {
     }
 
     dependencies {
-
-        implementation platform(boms.grpc),
-                platform(boms.guava),
-                platform(boms.protobuf),
-                platform(boms.zipkin_reporter)
+        configurations.all {
+            if (it.name.endsWith('Classpath')) {
```

Comment:
One usually does not add anything directly to `XXClasspath` configuration. Why is this needed in this case?

Reply:
The classpath configurations are only for resolving dependencies, not publishing them. Since we want to use the BOMs for resolving, not publishing (as we see in the large thread on the other PR ;) ), it makes sense to add them to the classpath configurations instead.

This is the recommended approach by Gradle for this use case - https://github.com/gradle/gradle/issues/10861#issuecomment-576562961

Reply:
Isn't it enough to use `fromResolutionResult` as you did below?

Reply:
Nah, at first I thought so too but while working on this realized it's a different topic.

- `fromResolutionResult` fills in version numbers for dependencies resolved via bom. So for `guava` dependency, without it, it would be present but have no version number

- Moving platform from "dependency bag" configurations to classpath configurations. Ensures BOM imports are not added to the published POM which is a good thing to reduce the complexity of the POM, we want the BOMs to resolve the version numbers, not to publish them